### PR TITLE
enh(centreon-php): add service to the search of 'monitoring/hostgroups' endpoint

### PR DESCRIPTION
**Pull request by Yassir-BenBOUBKER**
for the bulk mode tasks ( MAP )
we need Service to be added to the Search (service.display_name) In Host Groups / service groups

especially since the Doc 
https://docs-api.centreon.com/api/centreon-web/23.04/#tag/Host-group/paths/~1monitoring~1hostgroups/

supports this feature but it doesn’t work for real in the API

### CODE REVIEW

File: centreon/src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 1]
```diff
@@ -357,6 +357,11 @@ public function findHostGroups(?int $hostId): array
             'poller.id' => 'h.instance_id'
         ];
 
+        // To allow to find host groups relating to Service information
+        $ServiceConcordanceArray = [
```
Comment (a-launois) [inline centreon/src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php:361]:
```suggestion
        $serviceConcordanceArray = [
```

File: centreon/src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 2]
```diff
@@ -365,6 +370,12 @@ public function findHostGroups(?int $hostId): array
             $hostGroupConcordanceArray = array_merge($hostGroupConcordanceArray, $hostConcordanceArray);
         }
 
+        $shouldJoinService = false;
+        if (count(array_intersect($searchParameters, array_keys($ServiceConcordanceArray))) > 0) {
+            $shouldJoinService = true;
+            $hostGroupConcordanceArray = array_merge($hostGroupConcordanceArray, $ServiceConcordanceArray);
```
Comment (a-launois) [inline centreon/src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php:376]:
```suggestion
        if (count(array_intersect($searchParameters, array_keys($serviceConcordanceArray))) > 0) {
            $shouldJoinService = true;
            $hostGroupConcordanceArray = array_merge($hostGroupConcordanceArray, $serviceConcordanceArray);
```

File: centreon/src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 3]
```diff
@@ -357,6 +357,11 @@ public function findHostGroups(?int $hostId): array
             'poller.id' => 'h.instance_id'
         ];
 
+        // To allow to find host groups relating to Service information
+        $ServiceConcordanceArray = [
+            'service.display_name' => 'srv.display_name',
+        ];
```
Comment (adr-mo) [inline centreon/src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]: PHP uses camelCase for varirable definitions
```suggestion
        $serviceConcordanceArray = [
            'service.display_name' => 'srv.display_name',
        ];
```

File: centreon/src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 4]
```diff
@@ -365,6 +370,12 @@ public function findHostGroups(?int $hostId): array
             $hostGroupConcordanceArray = array_merge($hostGroupConcordanceArray, $hostConcordanceArray);
         }
 
+        $shouldJoinService = false;
+        if (count(array_intersect($searchParameters, array_keys($ServiceConcordanceArray))) > 0) {
+            $shouldJoinService = true;
+            $hostGroupConcordanceArray = array_merge($hostGroupConcordanceArray, $ServiceConcordanceArray);
+        }
```
Comment (adr-mo) [inline centreon/src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]:
```suggestion
        if (count(array_intersect($searchParameters, array_keys($serviceConcordanceArray))) > 0) {
            $shouldJoinService = true;
            $hostGroupConcordanceArray = array_merge($hostGroupConcordanceArray, $serviceConcordanceArray);
        }
```

**Review by adr-mo — CHANGES_REQUESTED** [review]

**Review by a-launois — DISMISSED** [review]

**Review by adr-mo — DISMISSED** [review]

**Review by wtermellil — DISMISSED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by callapa — CHANGES_REQUESTED** [review]
Please update the API documentation

**Review by callapa — DISMISSED** [review]

File: centreon/src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 5]
```diff
@@ -1332,6 +1350,36 @@ public function findServiceGroups(): array
             'name' => 'sg.name'
         ];
 
+        // To allow to find service groups relating to host information
+        $hostConcordanceArray = [
+            'host.id' => 'h.host_id',
+            'host.name' => 'h.name',
+            'host.alias' => 'h.alias',
+            'host.address' => 'h.address',
+            'host.display_name' => 'h.display_name',
```
Comment (callapa) [inline centreon/src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php:1359]: Property not referenced in the API documentation

**Review by callapa — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**